packages/gatsby-plugin-vercel-builder/src/helpers/static.ts

```diff
@@ -10,14 +10,13 @@ export async function createStaticDir(prefix?: string) {
     'static',
     prefix ?? ''
   );
-  await ensureDir(dirname(targetDir));
+
   try {
-    await move(publicDir, targetDir);
+    await hardLinkDir(publicDir, [targetDir]);
   } catch (err: any) {
-    console.error(
-      `Failed to move "public" dir from "${publicDir}" to "${targetDir}". Copying instead.`,
-      err
+    console.error(err);
+    throw new Error(
+      `Failed to hardlink "public" dir files from "${publicDir}" to "${targetDir}". Copying instead.`
     );
```

Comment:
We can remove the `Copying instead` message:

```suggestion
    throw new Error(
      `Failed to hardlink "public" dir files from "${publicDir}" to "${targetDir}"`,
      { cause: err }
    );
```

Reply:
`cause` is only used in edge function handling, isn't it?

Reply:
`cause` was introduced in ES2022 and support was added to Node.js 16.9.0 (note 16.13.0 was LTS). I'm thinking it's harmless in older Node.js versions as the 2nd argument would be ignored. I've only tested Node.js >=16.13.0 and >=18.12.0. This is the perfect use case for `cause` which is why I suggested it.

Reply:
Ahh, that would nice to use, yeah.

I can't get typescript to be OK with it. I tried setting this package's tsconfig to be:

```
"target": "es2022",
"lib": ["es2022"],
```

But still get:

```
Expected 0-1 arguments, but got 2.ts(2554)
```

Also, if a user installs the CLI and uses node 14 or a lower node 16, this second param will be ignored. It's not the biggest of deals, but they'd miss out on that error detail.

Reply:
If we can leverage this, I'd be happy to do it in a follow-up PR

Reply:
That sounds good. If we're going to add `cause` here, there are probably dozens of other places too. We have a ticket VCCLI-254 to refactor all of the error handling code so that we do more throwing and less returning a numeric code from multiple levels deep.

Reply:
Updated the message: [`39a0744` (#9673)](https://github.com/vercel/vercel/pull/9673/commits/39a074449289c43ab48f895896a32b7d1b8b221c)